app/upload/views.py

```diff
@@ -72,3 +82,22 @@ def upload_document(service_id):
             'mimetype': mimetype,
         }
     ), 201
+
+
+def _upload_document_with_hashed_recipient_email(verification_email, service_id, file_data, mimetype):
+    hasher = PasswordHasher(
+        memory_cost=15360,
+        time_cost=2,
+        parallelism=1,
+        hash_len=16,
+        type=Type.ID
+    )
+
+    hashed_recipient_email = hasher.hash(verification_email)
+    document = document_store.put(
+        service_id,
+        file_data,
+        mimetype=mimetype,
+        metadata={"Hashed-Recipient-Email": hashed_recipient_email}
```

Comment:
Also, looks like I have slightly misled you here - sorry. This will need to be `hashed-recipient-email` - in lowercase - as S3 automatically converts the key name to lowercase.

Reply:
No worries, I changed it now ☺️